.ci/pitest.groovy

```diff
@@ -0,0 +1,459 @@
+ * @param flag command line argument flag to determine output format
+ * @return {@code 0} if comparison passes successfully
+ */
+private static int compareMutations(Set<Mutation> survivedMutations,
+                                    Set<Mutation> suppressedMutations,
+                                    String flag) {
+
+    final int exitCode
+    final Set<Mutation> survivedAndUnsuppressedMutations =
+            setDifference(survivedMutations, suppressedMutations)
+    final Set<Mutation> unnecessarySuppressions =
+            setDifference(suppressedMutations, survivedMutations)
+    if (survivedMutations == suppressedMutations) {
+        exitCode = 0
+    }
+    else if (survivedAndUnsuppressedMutations.isEmpty()
+            && !containsOnlyStableMutations(unnecessarySuppressions)) {
+        exitCode = 0
+    }
+    else {
+        if (!survivedAndUnsuppressedMutations.isEmpty()) {
+            println "Surviving mutation(s) found:"
+            survivedAndUnsuppressedMutations.each {
+                printMutation(flag, it)
+            }
+        }
+        if (!unnecessarySuppressions.isEmpty()) {
+            println "\nUnnecessary suppressed mutation(s) found:"
+            unnecessarySuppressions.each {
+                if (!it.unstable()) {
+                    printMutation(flag, it)
+                }
+            }
+        }
+        exitCode = 1
+    }
+    return exitCode
+}
+
+/**
+ * Whether a set contains only stable mutations.
+ *
+ * @param mutations A set of mutations
+ * @return {@code true} if a set contains only stable mutations
+ */
+private static boolean containsOnlyStableMutations(Set<Mutation> mutations) {
+    final Mutation unstableMutation = mutations.find {
+        return it.unstable()
+    }
+    return unstableMutation != null
+}
+
+/**
+ * Prints the mutation according to the nature of the flag.
+ *
+ * @param flag command line argument flag to determine output format
+ * @param mutation mutation to print
+ */
+private static void printMutation(String flag, Mutation mutation) {
+    if (flag != null) {
+        println mutation.toXmlString()
+    }
+    else {
+        println mutation
+    }
+}
+
+/**
+ * Determine the difference between 2 sets. The result is {@code setOne - setTwo}.
+ *
+ * @param setOne The first set in the difference
+ * @param setTwo The second set in the difference
+ * @return {@code setOne - setTwo}
+ */
+private static Set<Mutation> setDifference(final Set<Mutation> setOne,
+                                           final Set<Mutation> setTwo) {
+    final Set<Mutation> result = new HashSet<Mutation>(setOne)
+    result.removeIf(mutation -> setTwo.contains(mutation))
+    return result
+}
+
+/**
+ * A class to represent the XML {@code mutation} node.
+ */
+record Mutation(String sourceFile, String mutatedClass,
+                String mutatedMethod, String mutator,
+                String description, String lineContent, int lineNumber,
+                boolean unstable)
```

Comment:
`isUnstable`

Reply:
Done.

Reply:
Does groovy remove `is` when generating getters?

Reply:
I created another getter, though groovy documentation mentioned-

> Note the special naming convention for record getters. They are the same name as the field (rather than the often common JavaBean convention of capitalized with a "get" prefix).

Done.